### PR TITLE
IGNITE-9845 Implemented support for passing arguments for mutual SSL

### PR DESCRIPTION
**Pull request by akuznetsov-gridgain**

### CODE REVIEW

File: modules/web-console/web-agent/src/main/java/org/apache/ignite/console/agent/rest/RestExecutorPool.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    }
+
+    /** */
+    public RestResult sendRequest(
+        String name,
+        List<String> nodeURIs,
+        Map<String, Object> params,
+        Map<String, Object> headers
+    ) throws IOException {
+        RestExecutor restExec = pool.get(name);
```
Comment (nva) [inline modules/web-console/web-agent/src/main/java/org/apache/ignite/console/agent/rest/RestExecutorPool.java]: RestExecutor should be cached by token

File: modules/web-console/web-agent/src/main/java/org/apache/ignite/console/agent/rest/RestExecutorPool.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+            restExec = new RestExecutor(cfg);
+
+            pool.put(name, restExec);
+        }
+
+        return restExec.sendRequest(nodeURIs, params, headers);
+    }
+
+    /** {@inheritDoc} */
+    @Override public void close() {
```
Comment (nva) [inline modules/web-console/web-agent/src/main/java/org/apache/ignite/console/agent/rest/RestExecutorPool.java]: Should be closed by token reset

File: modules/web-console/web-agent/src/main/java/org/apache/ignite/console/agent/rest/RestExecutorPool.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    /** */
+    public RestResult sendRequest(
+        String name,
+        List<String> nodeURIs,
+        Map<String, Object> params,
+        Map<String, Object> headers
+    ) throws IOException {
+        RestExecutor restExec = pool.get(name);
+
+        if (restExec == null) {
```
Comment (nva) [inline modules/web-console/web-agent/src/main/java/org/apache/ignite/console/agent/rest/RestExecutorPool.java]: Non atomic putIfAbsent

File: modules/web-console/web-agent/src/main/java/org/apache/ignite/console/agent/rest/RestExecutor.java [hunk 4]
```diff
@@ -94,8 +124,51 @@ public RestExecutor() {
...
+
+                    TrustManagerFactory tmf = TrustManagerFactory.getInstance("SunX509");
+                    tmf.init(trustKeyStore);
+
+                    TrustManager[] trustMgrs = tmf.getTrustManagers();
+
+                    X509TrustManager trustMgr = (X509TrustManager)Arrays.stream(trustMgrs)
+                        .filter(tm -> tm instanceof X509TrustManager)
+                        .findFirst()
+                        .orElseGet(null);
```
Comment (nva) [inline modules/web-console/web-agent/src/main/java/org/apache/ignite/console/agent/rest/RestExecutor.java]: use .orElseThrow

File: modules/web-console/web-agent/src/main/java/org/apache/ignite/console/agent/AgentConfiguration.java [hunk 5]
```diff
@@ -97,23 +97,23 @@
 
     /** */
     @Parameter(names = {"-cs", "--client-certificate"},
```
Comment (nva) [inline modules/web-console/web-agent/src/main/java/org/apache/ignite/console/agent/AgentConfiguration.java]: Also, please, add client certificate, client password for web console backend.

Comment (akuznetsov-gridgain) [reply modules/web-console/web-agent/src/main/java/org/apache/ignite/console/agent/AgentConfiguration.java]: Done

File: modules/web-console/frontend/webpack/webpack.dev.js [hunk 6]
```diff
@@ -23,9 +23,15 @@ const commonCfg = require('./webpack.common');
 
 const MiniCssExtractPlugin = require('mini-css-extract-plugin');
 
+const backendProtocol = process.env.BACKEND_PROTOCOL || 'http';
 const backendPort = process.env.BACKEND_PORT || 3000;
-const devServerPort = process.env.PORT || 9000;
 const devServerHost = process.env.HOST || '0.0.0.0';
+const devServerPort = process.env.PORT || 9000;
+
+console.log(`BACKEND_PROTOCOL: ${backendProtocol}`);
```
[the remark this answers is not in the document]
Comment (akuznetsov-gridgain) [reply modules/web-console/frontend/webpack/webpack.dev.js]: Reworked.

File: modules/web-console/assembly/README.txt [hunk 7]
```diff
@@ -90,9 +111,10 @@ In case of non GMail SMTP server it may require to change options in "settings.j
...
 1. On Windows check that MongoDB is not blocked by Antivirus/Firewall/Smartscreen.
-2. Root permission is required to bind to 80 port under Mac OS X and Linux, but you may always start Web Console on another port if you don't have such permission.
+2. Root permission is required to bind to 80 port under macOS and Linux, but you may always start Web Console
+   on another port if you don't have such permission.
 3. For extended debug output start Web Console as following:
-	On Linux execute command in terminal: `DEBUG=mongodb-* ./ignite-web-console-linux`
-	On Windows execute two commands in terminal:
-		`SET DEBUG=mongodb-*`
-		`ignite-web-console-win.exe`
+     On macOS/Linux execute command in terminal: `DEBUG=mongodb-* ./ignite-web-console-linux`
```
[the remark this answers is not in the document]
Comment (akuznetsov-gridgain) [reply modules/web-console/assembly/README.txt]: Fixed.

File: modules/web-console/web-agent/src/main/java/org/apache/ignite/console/agent/AgentUtils.java [hunk 8]
```diff
@@ -186,9 +208,149 @@ public static Object toJSON(Object obj) {
...
+            .orElseThrow(() -> new IllegalStateException("X509TrustManager manager not found"));
+    }
+
+    /**
+     * Parse comma-separated string to array.
+     *
+     * @param s String to parse.
+     * @return Array of strings or {@code null}.
+     */
+    @Nullable private static String[] parseArray(String s) {
```
[the remark this answers is not in the document]
Comment (akuznetsov-gridgain) [reply modules/web-console/web-agent/src/main/java/org/apache/ignite/console/agent/AgentUtils.java]: Agree, fixed

File: modules/web-console/web-agent/src/main/java/org/apache/ignite/console/agent/AgentUtils.java [hunk 9]
```diff
@@ -186,9 +208,149 @@ public static Object toJSON(Object obj) {
...
+        String trustStorePwd,
+        String cipherSuites
+    ) throws GeneralSecurityException, IOException {
+        boolean trustAll = Boolean.getBoolean("trust.all");
+        boolean hasKeyStore = keyStore != null;
+        boolean hasTrustStore = trustStore != null;
+
+        if (trustAll || hasKeyStore || hasTrustStore) {
+            if (trustAll && hasTrustStore) {
+                log.warn("Options contains both '--XXXX-trust-store' and '-Dtrust.all=true'. " +
```
[the remark this answers is not in the document]
Comment (akuznetsov-gridgain) [reply modules/web-console/web-agent/src/main/java/org/apache/ignite/console/agent/AgentUtils.java]: Reworked.

File: modules/web-console/web-agent/src/main/java/org/apache/ignite/console/agent/AgentUtils.java [hunk 10]
```diff
@@ -186,9 +208,149 @@ public static Object toJSON(Object obj) {
...
+     */
+    public static void initSsl(
+        OkHttpClient.Builder builder,
+        String keyStore,
+        String keyStorePwd,
+        String trustStore,
+        String trustStorePwd,
+        String cipherSuites
+    ) throws GeneralSecurityException, IOException {
+        boolean trustAll = Boolean.getBoolean("trust.all");
```
[the remark this answers is not in the document]
Comment (akuznetsov-gridgain) [reply modules/web-console/web-agent/src/main/java/org/apache/ignite/console/agent/AgentUtils.java]: Reworked.

File: modules/web-console/web-agent/src/main/java/org/apache/ignite/console/agent/AgentUtils.java [hunk 11]
```diff
@@ -206,4 +368,69 @@ public static X509TrustManager trustManager() {
             }
         };
     }
+
+    /**
+     * SSL socket factory that configure socket ciphers.
+     */
+    private static class SSLSocketFactoryWithCiphers extends SSLSocketFactory {
```
[the remark this answers is not in the document]
Comment (akuznetsov-gridgain) [reply modules/web-console/web-agent/src/main/java/org/apache/ignite/console/agent/AgentUtils.java]: Fixed.

File: modules/web-console/web-agent/src/main/java/org/apache/ignite/console/agent/AgentLauncher.java [hunk 12]
```diff
@@ -318,25 +316,37 @@ public static void main(String[] args) throws Exception {
...
-            log.info("Trust to all certificates mode is enabled.");
+        if (
+            Boolean.getBoolean("trust.all") ||
+            cfg.serverKeyStore() != null ||
+            cfg.serverTrustStore() != null
+        ) {
+            OkHttpClient.Builder builder = new OkHttpClient.Builder();
 
-            SSLContext ctx = SSLContext.getInstance("TLS");
+            AgentUtils.initSsl(builder,
```
[the remark this answers is not in the document]
Comment (akuznetsov-gridgain) [reply modules/web-console/web-agent/src/main/java/org/apache/ignite/console/agent/AgentLauncher.java]: Refactored.

File: modules/web-console/web-agent/src/test/java/org/apache/ignite/console/agent/rest/RestExecutorSelfTest.java [hunk 13]
```diff
@@ -0,0 +1,335 @@
...
+        cfg.setIgniteInstanceName(UUID.randomUUID().toString());
+
+        return cfg;
+    }
+
+    /**
+     * @param jettyCfg Name of file with Jetty XML config.
+     * @return Node configuration with enabled SSL for REST.
+     */
+    private IgniteConfiguration sslNodeConfiguration(String jettyCfg) {
```
Comment (nva) [inline modules/web-console/web-agent/src/test/java/org/apache/ignite/console/agent/rest/RestExecutorSelfTest.java]: Please rename. You only set jetty configuration path in this method.

**Pull request closed without merging**